sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubAsyncClient.java

```diff
@@ -32,9 +34,13 @@
     private final boolean isSharedConnection;
     private final Runnable onClientClose;
     private final TracerProvider tracerProvider;
+    private final boolean isIdempotentPartitionPublishing;
+    private final Map<String, PartitionPublishingState> initialPartitionPublishingStates;
 
     EventHubAsyncClient(EventHubConnectionProcessor connectionProcessor, TracerProvider tracerProvider,
-        MessageSerializer messageSerializer, Scheduler scheduler, boolean isSharedConnection, Runnable onClientClose) {
+        MessageSerializer messageSerializer, Scheduler scheduler, boolean isSharedConnection, Runnable onClientClose,
```

Comment:
We should have created an internal options bag before.. but it might be worthwhile because as we add to the constructor, it'll break existing classes.
EventHubAsyncClientOptions or something that has all these optional things (ie. isSharedConnection, isIdempotent, etc)

Reply:
I'd like to create an issue to do this internal refactor separately but not in this PR.